daemon/anonymous-statistics.sh

```diff
@@ -0,0 +1,146 @@
+KERNEL_VERSION="$(uname -r)"
+ARCHITECTURE="$(uname -m)"
+
+# -------------------------------------------------------------------------------------------------
+# detect the virtualization
+
+VIRTUALIZATION="unknown"
+VIRT_DETECTION="none"
+
+if [ ! -z "$(command -v systemd-detect-virt 2>/dev/null)" ]
+then
+    VIRTUALIZATION="$(systemd-detect-virt)"
+    VIRT_DETECTION="systemd-detect-virt"
+else
+	IFS='(, ' read -r process _ </proc/1/sched
+	if [ "${process}" == "netdata" ]; then
+	   VIRTUALIZATION="container"
+	   VIRT_DETECTION="pid"
+	fi
+
+    # ubuntu and debian supply /bin/running-in-container
+    # https://www.apt-browse.org/browse/ubuntu/trusty/main/i386/upstart/1.12.1-0ubuntu4/file/bin/running-in-container
+	if /bin/running-in-container >/dev/null 2>&1 ; then
+		VIRTUALIZATION="container"
+		VIRT_DETECTION="/bin/running-in-container"
+	fi
+
+    # lxc sets environment variable 'container'
+    [ ! -z "${container}" ] &&
+        VIRTUALIZATION="lxc" &&
+        VIRT_DETECTION="containerenv"
+
+    # docker creates /.dockerenv
+    # http://stackoverflow.com/a/25518345
+    [ -f "/.dockerenv" ] &&
+        VIRTUALIZATION="docker" &&
+        VIRT_DETECTION="dockerenv"
+fi
+
+# -------------------------------------------------------------------------------------------------
+# check netdata version
+
+if [ -z "${NETDATA_VERSION}" ]; then
+    NETDATA_VERSION="$(netdata -V 2>&1 | cut -d ' ' -f 2)"
+fi
+
+# -------------------------------------------------------------------------------------------------
+# check netdata unique id
+
+if [ -z "${NETDATA_REGISTRY_UNIQUE_ID}" ]; then
+    for dir in "/var/lib/netdata" "$(netdata -W get global 'lib directory' '/var/lib/netdata' 2>/dev/null)" "/opt/netdata/var/lib/netdata"; do
+        if [ -f "${dir}/registry/netdata.public.unique.id" ]; then
+            NETDATA_REGISTRY_UNIQUE_ID="$(cat "${dir}/registry/netdata.public.unique.id")"
+            break
+        fi
+    done
+fi
+
+# -------------------------------------------------------------------------------------------------
+# send the anonymous statistics to GA
+ACTION_DATA=$(echo "${ACTION_DATA}" | tr '"' "'")
+
+payload="&v=1&an=netdata&aip=1&t=pageview&_s=1&ds=statistics&tid=UA-64295674-3&cid=${NETDATA_REGISTRY_UNIQUE_ID}&t=event&av=${NETDATA_VERSION}&ec=${ACTION}&ea=${ACTION_RESULT}&dh=netdata-agent.my-netdata.io&dp=%2Fstatistics"
+
+curl -X POST -Ss --max-time 2 \
+--data-raw "${payload}" \
+--data-urlencode "el=${ACTION_DATA}" \
+--data-urlencode "cd1=${NAME}" \
+--data-urlencode "cd2=${ID}" \
+--data-urlencode "cd3=${ID_LIKE}" \
+--data-urlencode "cd4=${VERSION}" \
+--data-urlencode "cd5=${VERSION_ID}" \
+--data-urlencode "cd6=${OS_DETECTION}" \
+--data-urlencode "cd7=${KERNEL_NAME}" \
+--data-urlencode "cd8=${KERNEL_VERSION}" \
+--data-urlencode "cd9=${ARCHITECTURE}" \
+--data-urlencode "cd10=${VIRTUALIZATION}" \
+--data-urlencode "cd11=${VIRT_DETECTION}" \
+"https://www.google-analytics.com/collect" >/dev/null 2>&1
```

Comment:
Please make it more readable. It shouldn't be this hard to do a code review of this reqest as GA docs are pretty clear. However this takes too much time due to jumping from one section to another.

To make it more readable please:
- stop mixing `data-raw` with `data-urlencode` and stick to one of those.
- cleanup ordering of parameters and group them in the same way as they are grouped in GA docs: https://developers.google.com/analytics/devguides/collection/protocol/v1/parameters

Reply:
> stop mixing data-raw with data-urlencode and stick to one of those.

No need to url-encode what certainly doesn't require url-encoding just to make it pretty. Did the rest.